internal/sail/client/client.go

```diff
@@ -69,11 +88,11 @@ func (s *SailClient) broadcast(ctx context.Context, view webview.View) {
 
 	err := s.conn.WriteJSON(view)
 	if err != nil {
-		logger.Get(ctx).Infof("broadcast(%s): %v", s.addr, err)
+		logger.Get(s.ctx).Infof("broadcast(%s): %v", s.addr, err)
```

Comment:
i'm pretty sure this context won't have a logger attached, so this will be an error

Reply:
as a rule of thumb, "Context" is for passing along information about the current call context, like tracing information.

If you're storing a context on an object that's alive for the entire program, you're probably using it wrong. In all of the places you're using it in this file, it'd be better to pass a context in

Reply:
yeah this was all an artifact of trying to get a ctx into these functions that were being called from inside the hud http handler.

Reply:
fwiw, there should be a req.Context() that you can get in the http handler

Reply:
oh, amazing!